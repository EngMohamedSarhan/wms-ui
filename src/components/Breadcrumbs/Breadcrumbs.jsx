import PropTypes from "prop-types";
import React, { Fragment } from "react";

import BackButton from "../BackButton/BackButton";
import RouteChip from "../RouteChip/RouteChip";

const Breadcrumbs = ({ label, icon, sx }) => (
  <Fragment>
    <RouteChip icon={icon} label={label} sx={sx} />
    <BackButton sx={{ ml: 2 }} />
  </Fragment>
);

Breadcrumbs.propTypes = {
  label: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  sx: PropTypes.object,
};

export default Breadcrumbs;
