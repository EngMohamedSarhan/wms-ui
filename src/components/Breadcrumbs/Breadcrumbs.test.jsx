/* eslint-disable no-undef */
import { TextSnippet } from "@mui/icons-material";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import { Router } from "react-router-dom";
import ShallowRenderer from "react-shallow-renderer";

import Breadcrumbs from "./Breadcrumbs";

describe("Breadcrumbs Component", () =>
  test("Breadcrumbs rendering", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Breadcrumbs label="test" icon={TextSnippet} />
      </Router>
    );
    const chip = getByTestId("route-chip");
    const backButton = getByTestId("navigate-home");
    expect(chip).toBeInTheDocument();
    expect(backButton).toBeInTheDocument();
  }));

describe("Breadcrumbs Component", () =>
  test("Breadcrumbs snapshot", () => {
    const renderer = new ShallowRenderer();
    const history = createMemoryHistory({ initialEntries: ["/lpn-date-check"] });
    renderer.render(
      <Router location={{ ...history.location, key: "1" }}>
        <Breadcrumbs label="test" icon={TextSnippet} />
      </Router>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
