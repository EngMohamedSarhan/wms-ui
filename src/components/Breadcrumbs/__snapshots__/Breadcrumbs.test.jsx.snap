// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Component Breadcrumbs snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <Breadcrumbs
      icon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
      label="test"
    />
  </Context.Provider>
</Context.Provider>
`;

exports[`Breadcrumbs snapshot Breadcrumbs snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <Breadcrumbs
      icon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
      label="test"
    />
  </Context.Provider>
</Context.Provider>
`;
